source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/connect/ServerTypeSelectControl.java

```diff
@@ -245,4 +382,42 @@ public URI getServerURI() {
             return serverURI;
         }
     }
+
+    private class PropsalProvider implements IContentProposalProvider {
+        private String[] proposals = new String[0];

```

Comment:
Wouldn't it more convenient to use List<String> here? We might make proposals final.